tools/qr-generator/src/generator/codes.js

```diff
@@ -0,0 +1,20 @@
+const SafeChars = [...'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789']
+
+const generateCode = (length) =>
+  Array.from({ length })
+    .map(() => SafeChars[Math.floor(Math.random() * SafeChars.length)])
+    .join('')
+
+export const generateUniqueCodes = ({ count, prefix, length }) => {
```

Comment:
Is this more understandable?

```
export const generateUniqueCodes = ({ count, prefix, length }) => {
  const codes = new Set();

  let attempt = 0;
  while (codes.size < count) {
    attempt++;

    if (attempt > count * 100) {
      throw new Error('Failed to generate enough unique codes, consider increasing the code length.');
    }

    const newCode = prefix + generateCode(length - prefix.length);
    codes.add(newCode);
  }

  return [...codes];
};
```

Reply:
Yes, it's a lot better, I tried way to much to avoid the while loop